php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java

```diff
@@ -0,0 +1,258 @@
+                return new Dimension(3, 3); // Y-unimportant -> gridlayout will stretch it
+            }
+        };
+        separator.setBorder(BorderFactory.createEmptyBorder(1, 1, 1, 1));
+
+        JPanel panel = new JPanel(new BorderLayout());
+        panel.add(separator, BorderLayout.WEST);
+        panel.add(cell);
+        return panel;
+    }
+
+    //~ Inner classes
+    private static final class PhpVersionLabel extends JLabel {
+
+        private static final Insets NULL_INSETS = new Insets(0, 0, 0, 0);
+        private static final PhpVersionLabel INSTANCE = new PhpVersionLabel();
+        private static final long serialVersionUID = -3407578574821051662L;
+
+        static {
+            EditorRegistry.addPropertyChangeListener(propertyChangeEvent -> getInstance().updatePhpVersion());
+        }
+
+        private PhpVersionLabel() {
+            // show the version list when the label is clicked
+            addMouseListener(new MouseAdapterImpl());
+            initMinDimension(getPhpVersionStrings());
+        }
+
+        public static PhpVersionLabel getInstance() {
+            return INSTANCE;
+        }
+
+        @Override
+        public Point getToolTipLocation(MouseEvent event) {
+            // show the tooltip left side of the label
+            String toolTipText = getToolTipText(event);
+            JToolTip jToolTip = new JToolTip();
+            jToolTip.setTipText(toolTipText);
+            return new Point(-5 - jToolTip.getPreferredSize().width, 0);
+        }
+
+        private Collection<String> getPhpVersionStrings() {
+            Collection<String> phpVersionStrings = new ArrayList<>();
+            for (PhpVersion phpVersion : PhpVersion.values()) {
+                phpVersionStrings.add(phpVersion.getDisplayName());
+            }
+            return phpVersionStrings;
+        }
+
+        private void initMinDimension(Iterable<? extends String> maxStrings) {
+            FontMetrics fontMetrics = getFontMetrics(getFont());
+            int minWidth = 0;
+            for (String string : maxStrings) {
+                minWidth = Math.max(minWidth, fontMetrics.stringWidth(string));
+            }
+            Border border = getBorder();
+            Insets insets = (border != null) ? border.getBorderInsets(this) : NULL_INSETS;
+            minWidth += insets.left + insets.right;
+            int minHeight = fontMetrics.getHeight() + insets.top + insets.bottom;
+            setMinimumSize(new Dimension(minWidth, minHeight));
+            setPreferredSize(new Dimension(minWidth, minHeight));
+        }
+
+        private void updatePhpVersion() {
+            assert EventQueue.isDispatchThread();
+            final JTextComponent comp = EditorRegistry.focusedComponent();
+            if (comp != null) {
+                showPhpVersion(comp.getDocument());
+            } else {
+                clear();
+            }
+        }
+
+        private void showPhpVersion(Document document) {
```

Comment:
This is done in the UI thread... hmm, hopefully it will be fast, ideally always :)

Reply:
(Otherwise some kind of "Loading..." would need to be added.)

Reply:
e.g. `setText(Bundle.loading_message())`?

Reply:
I meant - if this is slow and blocking the UI, we would need to:
- show Loading text here
- start a bg thread which would do the work
- once finished, the Loading text would be replaced with the PHP version

Is this what you meant? Or...? Thanks!

Reply:
I see. Thanks!

Reply:
Perfect, thank you!